main.go

```diff
@@ -68,11 +69,20 @@ func main() {
 	// externalAuth - Default to user/password but if no password is supplied then will automagically set to true
 	externalAuth := false
 
-	vaultID, useVault := os.LookupEnv("OCI_VAULT_ID")
-	if useVault {
+	ociVaultID, useOciVault := os.LookupEnv("OCI_VAULT_ID")
+	if useOciVault {
 
-		logger.Info("OCI_VAULT_ID env var is present so using OCI Vault", "vaultOCID", vaultID)
-		password = vault.GetVaultSecret(vaultID, os.Getenv("OCI_VAULT_SECRET_NAME"))
+		logger.Info("OCI_VAULT_ID env var is present so using OCI Vault", "vaultOCID", ociVaultID)
+		password = ocivault.GetVaultSecret(ociVaultID, os.Getenv("OCI_VAULT_SECRET_NAME"))
+	}
+
+	azVaultID, useAzVault := os.LookupEnv("AZ_VAULT_ID")
```

Comment:
Typically this would be implemented using a provider pattern, e.g., a database authentication provider that OCI, Azure, and any other authentication providers would implement. Each provider implements the same interface, and there is a enum parameter to select between the chosen provider.

Reply:
azure does not have different apis like oci does, it just automagically works it out :)

Reply:
oh, scratch that, i see what you mean

Reply:
i can create an issue to refactor that later

Reply:
Do we need to get this done now rather than do it right the first go-round?